compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -1177,4 +1177,16 @@ bool hasFusedLeadingOp(linalg::LinalgOp rootOp) {
   });
 }
 
+std::optional<VscaleRange>
+getDefaultVscaleRange(IREE::HAL::ExecutableTargetAttr targetAttr) {
+  if (isAArch64(targetAttr)) {
+    // On AArch64 the scalable vector length will always be between 128-bit and
+    // 2048-bit. This works out as a vscale range of 1 to 16. See:
+    // https://developer.arm.com/Architectures/Scalable%20Vector%20Extensions
+    return VscaleRange{1, 16};
```

Comment:
Every time I see this, I'd like to suggest to use designated initializers. Then I realized that it is a c++20 feature. :p